plugins/inputs/dpdk/dpdk.go

```diff
@@ -0,0 +1,240 @@
+  ## This includes both writing to and reading from socket.
+  ## Setting the value to 0 disables the timeout (not recommended)
+  # socket_access_timeout = "200ms"
+
+  ## Enables telemetry data collection for selected device types.
+  ## Adding "ethdev" enables collection of telemetry from DPDK NICs (stats, xstats, link_status).
+  ## Adding "rawdev" enables collection of telemetry from DPDK Raw Devices (xstats).
+  device_types = ["ethdev", "rawdev"]
+
+  ## List of custom, application-specific telemetry commands to query 
+  ## For e.g. L3 Forwarding with Power Management Sample Application this could be: 
+  ##   additional_commands = ["/l3fwd-power/stats"]
+  # additional_commands = []
+
+  ## Allows turning off collecting data for individual "ethdev" commands.
+  ## Remove "/ethdev/link_status" from list to start getting link status metrics.
+  [inputs.dpdk.ethdev]
+    exclude_commands = ["/ethdev/link_status"]
+
+  ## When running multiple instances of the plugin it's recommended to add a unique tag to each instance to identify
+  ## metrics exposed by an instance of DPDK application. This is useful when multiple DPDK apps run on a single host.  
+  ##  [inputs.dpdk.tags]
+  ##    dpdk_instance = "my-fwd-app"
+`
+	defaultPathToSocket        = "/var/run/dpdk/rte/dpdk_telemetry.v2"
+	defaultAccessTimeout       = config.Duration(200 * time.Millisecond)
+	maxCommandLength           = 56
+	maxCommandLengthWithParams = 1024
+	dpdkPluginName             = "dpdk"
+	ethdevListCommand          = "/ethdev/list"
+	rawdevListCommand          = "/rawdev/list"
+)
+
+type dpdk struct {
+	SocketPath         string          `toml:"socket_path"`
+	AccessTimeout      config.Duration `toml:"socket_access_timeout"`
+	DeviceTypes        []string        `toml:"device_types"`
+	EthdevConfig       ethdevConfig    `toml:"ethdev"`
+	AdditionalCommands []string        `toml:"additional_commands"`
+	Log                telegraf.Logger `toml:"-"`
+	connector          *dpdkConnector  `toml:"-"`
+	rawdevCommands     []string        `toml:"-"`
+	ethdevCommands     []string        `toml:"-"`
+}
+
+type ethdevConfig struct {
+	EthdevExcludeCommands []string `toml:"exclude_commands"`
+}
+
+func init() {
+	inputs.Add(dpdkPluginName, func() telegraf.Input {
+		dpdk := &dpdk{
+			SocketPath:     defaultPathToSocket,
+			AccessTimeout:  defaultAccessTimeout,
+			rawdevCommands: []string{"/rawdev/xstats"},
+			ethdevCommands: []string{"/ethdev/stats", "/ethdev/xstats", "/ethdev/link_status"},
+		}
+		return dpdk
+	})
+}
+
+func (dpdk *dpdk) SampleConfig() string {
+	return dpdkSampleConfig
+}
+
+func (dpdk *dpdk) Description() string {
+	return dpdkDescription
+}
+
+// Performs validation of all parameters from configuration
+func (dpdk *dpdk) Init() error {
+	if dpdk.SocketPath == "" {
+		dpdk.SocketPath = defaultPathToSocket
+		dpdk.Log.Infof("using default '%v' path for socket_path", defaultPathToSocket)
```

Comment:
Please make this at most a `Debugf` or remove altogether. Usually there is more than one plugin configured with telegraf and cluttering the log with messages like this is not helpful.

Reply:
Good point, changed to `Debugf`.